internal/backend/secrets/aws.go

```diff
@@ -0,0 +1,108 @@
+package secrets
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+	"go.uber.org/zap"
+)
+
+type awsSecrets struct {
+	client  *secretsmanager.Client
+	logger  *zap.Logger
+	timeout time.Duration
+}
+
+// NewAWSSecrets initializes a client connection to AWS Secrets Manager.
+func NewAWSSecrets(l *zap.Logger, c *Config) (Secrets, error) {
+	cfg, err := config.LoadDefaultConfig(context.Background())
+	if err != nil {
+		l.Error("AWS config initialization", zap.Error(err))
+		return nil, err
+	}
+
+	client := secretsmanager.NewFromConfig(cfg)
+	return &awsSecrets{client: client, logger: l, timeout: c.Timeout}, nil
+}
+
+// The data size limit is 64 KiB, according to this link:
+// https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_CreateSecret.html
+func (s *awsSecrets) Set(ctx context.Context, scope, name string, data map[string]string) error {
+	ctx, cancel := limitContext(ctx, s.timeout)
+	defer cancel()
+
+	d, err := json.MarshalIndent(data, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	_, err = s.client.CreateSecret(ctx, &secretsmanager.CreateSecretInput{
+		Name:         aws.String(secretPath(scope, name)),
+		SecretString: aws.String(string(d)),
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *awsSecrets) Get(ctx context.Context, scope, name string) (map[string]string, error) {
+	ctx, cancel := limitContext(ctx, s.timeout)
+	defer cancel()
+
+	sec, err := s.client.GetSecretValue(ctx, &secretsmanager.GetSecretValueInput{
+		SecretId: aws.String(secretPath(scope, name)),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	data := make(map[string]string)
```

Comment:
I think secret can be main 
Or just plain string
Do we support it? Want to support it?

Binary is out of scope

Reply:
Not sure what you mean by "main". Plain strings are technically possible, but secrets managers are usually used to save JSON (`map[string]any`) and not just arbitrary strings, and we use a subset of JSON: `map[string]string`. Specifically, we store multiple strings per connection, not just a single one. That's why the API is defined this way, and the AWS implementation simply conforms to that.

Reply:
I meant json, the auto correct changed it 

So this implementation is for connections only?
No other secrets?

Reply:
See https://github.com/autokitteh/autokitteh/pull/69#discussion_r1520067876